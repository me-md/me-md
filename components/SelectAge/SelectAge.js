import React, { Component } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Item, Input } from 'native-base';

export default class SelectAge extends Component {
  render() {
    return (
      <View style={styles.selectAgeContainer}>
        <LinearGradient
          style={styles.gradientBackground}
          colors={['#004EFF', '#88CCF1']}
        >
          <Text style={styles.selectAgeHeader}>Age</Text>
          <Item style={styles.selectAgeInput}>
            <Input
              placeholder='Enter your age'
              style={styles.selectAgeInputText}
            />
          </Item>
        </LinearGradient>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  selectAgeContainer: {
    flex: 1
  },
  gradientBackground: {
    alignItems: 'center',
    flex: 1
  },
  selectAgeHeader: {
    color: '#fff',
    flex: 1,
    fontSize: 48,
    fontWeight: 'bold'
  },
  selectAgeInput: {
    color: '#FFF'
  }
});
