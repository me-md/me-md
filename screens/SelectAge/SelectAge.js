import React, { useState } from 'react';
import { Dimensions, StyleSheet, Text, View } from 'react-native';
import { LinearGradient } from 'expo-linear-gradient';
import { Item, Input } from 'native-base';
import { Entypo, MaterialIcons } from '@expo/vector-icons';

const height = Dimensions.get('window').height;

export default function SelectAge({ navigation }) {
  const sex = navigation.state.params.sex;
  const [age, setAge] = useState('');

  return (
    <View style={styles.container}>
      <LinearGradient
        style={styles.gradientBackground}
        colors={['#004EFF', '#88CCF1']}
      >
        <View style={styles.contentContainer}>
          <Entypo
            name='chevron-thin-up'
            size={36}
            color='white'
            onPress={() => navigation.goBack()}
          />
          <View style={styles.childContainer}>
            <Text style={styles.title}>Age</Text>
            <MaterialIcons name='cake' size={80} color='white' />
            <Item style={styles.input}>
              <Input
                placeholder='35'
                placeholderTextColor="#919191"
                style={styles.input}
                onChangeText={text => setAge(text)}
              />
            </Item>
          </View>
          {age !== '' ? (<Entypo
            name='chevron-thin-down'
            size={36}
            color='white'
            onPress={() => navigation.push('Location', { sex, age })}
          />) : (<Entypo
            name='chevron-thin-down'
            size={36}
            color='white'
            style={{ opacity: 0 }}
          />)}
        </View>
      </LinearGradient>
    </View>
  );
}

const styles = StyleSheet.create({
  contentContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-between',
    marginBottom: height * 0.06,
    marginTop: height * 0.06,
  },
  container: {
    flex: 1
  },
  gradientBackground: {
    alignItems: 'center',
    flex: 1
  },
  title: {
    color: '#fff',
    fontSize: 48,
    fontWeight: 'bold'
  },
  input: {
    color: '#FFF',
    fontSize: 36,
    fontWeight: 'bold',
    textAlign: 'center',
    width: '20%'
  },
  childContainer: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-evenly',
    marginBottom: height * 0.30
  }
});
