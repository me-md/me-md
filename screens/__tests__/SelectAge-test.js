import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import SelectAge from '../../screens/SelectAge/SelectAge';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));

describe('SelectAge', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it(`should match a snapshot`, () => {
    const mockNavigation = {

      state: {
        params: {
          sex: 'male'
        }
      }
    };

    const tree = renderer.create(<SelectAge navigation={mockNavigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
