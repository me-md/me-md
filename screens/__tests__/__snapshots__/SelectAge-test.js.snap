// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectAge should match a snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278210303,
          4287155441,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "marginBottom": 80.03999999999999,
          "marginTop": 80.03999999999999,
        }
      }
    >
      <Text />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "space-evenly",
            "marginBottom": 400.2,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 48,
              "fontWeight": "bold",
            }
          }
        >
          Age
        </Text>
        <Text />
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomWidth": 1,
              "borderColor": "#D9D5DC",
              "borderLeftWidth": 0,
              "borderRadius": undefined,
              "borderRightWidth": 0,
              "borderTopWidth": 0,
              "borderWidth": undefined,
              "color": "#FFF",
              "flexDirection": "row",
              "fontSize": 36,
              "fontWeight": "bold",
              "marginLeft": 2,
              "opacity": 1,
              "textAlign": "center",
              "width": "20%",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            editable={true}
            onChangeText={[Function]}
            placeholder="35"
            placeholderTextColor="#919191"
            rejectResponderTermination={true}
            style={
              Array [
                Object {
                  "color": "#000",
                  "flex": 1,
                  "fontSize": 17,
                  "height": 50,
                  "paddingLeft": 5,
                  "paddingRight": 5,
                  "top": 1.5,
                },
                Object {
                  "color": "#FFF",
                  "fontSize": 36,
                  "fontWeight": "bold",
                  "textAlign": "center",
                  "width": "20%",
                },
              ]
            }
            underlineColorAndroid="rgba(0,0,0,0)"
          />
        </View>
      </View>
      <Text />
    </View>
  </ViewManagerAdapter_ExpoLinearGradient>
</View>
`;
